core/src/main/java/com/linecorp/armeria/internal/metric/RequestMetricSupport.java

```diff
@@ -65,9 +65,9 @@ public static void setup(RequestContext ctx, AttributeKey<Boolean> requestMetric
         }
         ctx.setAttr(requestMetricsSetKey, true);
 
-        ctx.log().addListener(log -> onRequest(log, meterIdPrefixFunction, server),
-                              RequestLogAvailability.REQUEST_HEADERS,
-                              RequestLogAvailability.REQUEST_CONTENT);
+        ctx.log()
+           .whenAvailable(RequestLogProperty.REQUEST_HEADERS, RequestLogProperty.REQUEST_CONTENT)
+           .thenAccept(log -> onRequest(log, meterIdPrefixFunction, server));
```

Comment:
I thought so initially but it seem to break the existing behavior/tests, mainly because not all services consume the request completely. I think we need to pick some essential properties. What else would be useful?

Reply:
Added some Javadoc to `MeterIfPrefixFunction` about what properties will be available.